src/elements/ifc-client-link.ts

```diff
@@ -0,0 +1,58 @@
+import { Client, EnvData } from '../client';
+const PATH_ATTR = 'path';
+
+/**
+ * A custom element responsible for rendering an anchor element, turning a path
+ * within the current application to a full top-level url.
+ * It will be registered as `ifc-client-link` by a call to [[registerCustomElementsClient]] and
+ * should not be created directly. Instead add a `<ifc-client-link>` element to your
+ * markup or use `document.createElement('ifc-client-link')` after calling
+ * [[registerCustomElementsClient]].
+ */
+
+export default class IfcClientLinkElement extends HTMLElement {
+  private _client: Client;
+  private _environmentData: EnvData;
+  private _assignedRoute: string;
+
+  /** @internal */
+  constructor() {
+    super();
+    this._client = new Client({
+      hostOrigin: 'http://localhost:8080'
+    });
+    this._environmentData = {
```

Comment:
I don't know how to get the `environmentalData` and `assignedRoute` to the component. Here I just hard coded some values.

Reply:
With access to a `Client` instance, you won't need the environment data explicitly, since you can just use `urlFromClientPath` (which you are doing already!)